### ISSUE TITLE
Unexpected output when looping over empty objects

### ISSUE THREAD

**Issue by planeiii-te** [mikefarah/yq#1613]
**Describe the bug**
Looping over an empty object still returns a list of elements somehow.

Version of yq: `4.33.1`
Operating system: `mac`
Installed via: `homebrew`

**Input Yaml**
meh.yml:
```yaml
other: thing
labels: {}
```

**Command**
The command you ran:

```shell
yq '.labels[] | select(tag != "!!str") | ">" + key + "< " + (key | line) + " " + tag' meh.yaml
```

**Actual behavior**

Consistently outputs `0` as the `line` number. The `key` is the empty string. This doesn't change even when `labels` is much lower down in the file.

```shell
>< 0 !!null
```

**Expected behavior**

```shell
# no output
```

**Comment by mikefarah** [comment]
Sorry - dropped off my brain. Yep that's a bug, fill fix next release

**Comment by mikefarah** [comment]
Fixed in 4.33.3
